pkg/operations/remove.go

```diff
@@ -42,9 +43,12 @@ func CleanupVM(vm *api.VM) error {
 	// TODO should this function return a proper error?
 	RemoveVMContainer(inspectResult)
 
-	// After removal is successful, remove the dm snapshots
-	if err := cleanup.DeactivateSnapshot(vm); err != nil {
-		return err
+	// After remove the VM container, and the SnapshotDev still there
+	if _, err := os.Stat(vm.SnapshotDev()); err == nil {
```

Comment:
The fact that we have to do this is a code-smell.
I have a feeling we need to look at how we store vm state.
It's a bit weird that we could run this Cleanup function and not know whether there is a container or snapshot to delete.

Reply:
indeed.
we need to take a look into the mechanism of the vm removal process.